runtime/src/bank.rs

```diff
@@ -3515,9 +3515,7 @@ impl Bank {
                             process_result = Err(e);
                         }
 
-                        if process_result.is_ok() {
-                            self.update_executors(executors);
-                        }
+                        self.update_executors(executors);
```

Comment:
I think we need to differentiate between a simple cache miss and an updated executor here. I think the reason that failed transactions didn't update the cache is because you could have updated a program (which updates the invoke context executors map) and then a later instruction fails. If this happens, we shouldn't put the updated program in the bank's executors cache

Reply:
Yeah, good point, this was the original reason we didn't cache failed txs.